lib/Elastica/Query/FunctionScore.php

```diff
@@ -69,19 +76,27 @@ public function setFilter(AbstractFilter $filter)
      *
      * @param string         $functionType   valid values are DECAY_* constants and script_score
      * @param array|float    $functionParams the body of the function. See documentation for proper syntax.
-     * @param AbstractFilter $filter         optional filter to apply to the function
+     * @param AbstractQuery  $filter         optional filter to apply to the function
      * @param float          $weight         function weight
      *
      * @return $this
      */
-    public function addFunction($functionType, $functionParams, AbstractFilter $filter = null, $weight = null)
```

Comment:
Should we rename the param the $query instead f $filter to make it more obvious? This would apply to all changes.